### PR TITLE
Introduce new APIRule version

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Allow multiple service per host with overrides
- Create conversion webhook and certificate management job
- Migrate config yamls to latest kubebuilder

**Related issues**
https://github.com/kyma-project/kyma/issues/9936

### CODE REVIEW

File: api/v1alpha1/apirule_conversion.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import (
+	"encoding/json"
+	"log"
+
+	"github.com/kyma-incubator/api-gateway/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts this CronJob to the Hub version (v1beta1).
```
Comment (cnvergence) [inline api/v1alpha1/apirule_conversion.go:11]: outdated comment, pls fix

File: api/v1alpha1/apirule_conversion.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	dst.ObjectMeta = src.ObjectMeta
+
+	host := *src.Spec.Service.Host
+	dst.Spec.Host = &host
+
+	return nil
+}
+
+// ConvertFrom converts this CronJob from the Hub version (v1beta1).
```
Comment (cnvergence) [inline api/v1alpha1/apirule_conversion.go:38]: outdated comment, pls fix

File: api/v1alpha1/apirule_conversion.go (hunk 1 above)
Comment (barchw) [reply api/v1alpha1/apirule_conversion.go]: Comment updated

File: api/v1alpha1/apirule_conversion.go (hunk 2 above)
Comment (barchw) [reply api/v1alpha1/apirule_conversion.go]: Comment updated

File: config/crd/bases/gateway.kyma-project.io_apirules.yaml [hunk 3]
```diff
@@ -56,17 +58,6 @@ spec:
                               keys vary per handler.
                             type: object
                             x-kubernetes-preserve-unknown-fields: true
-                            properties:
```
Comment (cnvergence) [inline config/crd/bases/gateway.kyma-project.io_apirules.yaml:59]: I think we need to restore this CRD validation for now

Comment (barchw) [reply config/crd/bases/gateway.kyma-project.io_apirules.yaml]: Forgot to readd it here, it's done in kyma resources

File: api/v1beta1/apirule_webhook.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+package v1beta1
+
+import (
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// log is for logging in this package.
+var apirulelog = logf.Log.WithName("apirule-resource")
```
Comment (cnvergence) [inline api/v1beta1/apirule_webhook.go:25]: apirulelog is not used

File: api/v1alpha1/apirule_conversion.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+// ConvertFrom converts this ApiRule from the Hub version (v1beta1).
+func (dst *APIRule) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1beta1.APIRule)
+	specData, err := json.Marshal(src.Spec)
+	if err != nil {
+		return err
+	}
+
+	json.Unmarshal(specData, &dst.Spec)
```
Comment (cnvergence) [inline api/v1alpha1/apirule_conversion.go:46]: let's check the error value here as well

File: api/v1alpha1/apirule_conversion.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+	statusData, err := json.Marshal(src.Status)
+	if err != nil {
+		return err
+	}
+
+	json.Unmarshal(statusData, &dst.Status)
```
Comment (cnvergence) [inline api/v1alpha1/apirule_conversion.go:53]: same here

File: api/v1alpha1/apirule_conversion.go [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+func (src *APIRule) ConvertTo(dstRaw conversion.Hub) error {
+	log.Default().Printf("dst host: %s", src.Name)
+	dst := dstRaw.(*v1beta1.APIRule)
+
+	specData, err := json.Marshal(src.Spec)
+	if err != nil {
+		return err
+	}
+
+	json.Unmarshal(specData, &dst.Spec)
```
Comment (cnvergence) [inline api/v1alpha1/apirule_conversion.go:21]: let's check the error value here

File: api/v1alpha1/apirule_conversion.go [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+	statusData, err := json.Marshal(src.Status)
+	if err != nil {
+		return err
+	}
+
+	json.Unmarshal(statusData, &dst.Status)
```
Comment (cnvergence) [inline api/v1alpha1/apirule_conversion.go:28]: same here

Comment (barchw) [reply api/v1alpha1/apirule_conversion.go]: Checking now

File: api/v1alpha1/apirule_conversion.go (hunk 7 above)
Comment (barchw) [reply api/v1alpha1/apirule_conversion.go]: Checking now

File: api/v1alpha1/apirule_conversion.go (hunk 6 above)
Comment (barchw) [reply api/v1alpha1/apirule_conversion.go]: Checking now

File: api/v1alpha1/apirule_conversion.go (hunk 5 above)
Comment (barchw) [reply api/v1alpha1/apirule_conversion.go]: Checking now

File: api/v1beta1/apirule_webhook.go (hunk 4 above)
Comment (barchw) [reply api/v1beta1/apirule_webhook.go]: removed

**Comment by cnvergence** [comment]
LGTM

**Review by cnvergence — APPROVED** [review]

**Pull request merged**